src/analyses/interval_domain.h

```diff
@@ -40,10 +40,15 @@ class interval_domaint:public ai_domain_baset
     const ai_baset &ai,
     const namespacet &ns) const final;
 
+  bool join(const interval_domaint &b);
```

Comment:
Why is "join" being introduced here, i.e., why not implement this functionality in `merge`?